vllm/entrypoints/openai/serving_chat.py

```diff
@@ -252,7 +267,26 @@ async def chat_completion_stream_generator(
                                     model=model_name)
                                 if (request.stream_options and
                                         request.stream_options.include_usage):
-                                    chunk.usage = None
+                                    if (request.stream_options.
+                                            continuous_usage_stats
+                                            or res.outputs[i].finish_reason
+                                            is not None):
+                                        prompt_tokens = len(
+                                            res.prompt_token_ids)
+                                        completion_tokens = len(
+                                            res.outputs[i].token_ids)
+                                        usage = UsageInfo(
+                                            prompt_tokens=prompt_tokens,
+                                            completion_tokens=completion_tokens,
+                                            total_tokens=prompt_tokens +
+                                            completion_tokens,
+                                        )
+                                    if (request.stream_options.
+                                            continuous_usage_stats):
+                                        chunk.usage = usage
+                                    else:
+                                        chunk.usage = None
+
```

Comment:
Same comments apply here as above: (a) I don't think we need to check the finish reason since we are just echoing the prompt and (b) we can simplify the if/else logic in the same way as above.